rig-core/src/message.rs

```diff
@@ -0,0 +1,276 @@
+            "text/xml" => Some(DocumentMediaType::XML),
+            "application/x-javascript" | "text/x-javascript" => Some(DocumentMediaType::Javascript),
+            "application/x-python" | "text/x-python" => Some(DocumentMediaType::Python),
+            _ => None,
+        }
+    }
+
+    pub fn to_mime_type(&self) -> &'static str {
+        match self {
+            DocumentMediaType::PDF => "application/pdf",
+            DocumentMediaType::TXT => "text/plain",
+            DocumentMediaType::RTF => "text/rtf",
+            DocumentMediaType::HTML => "text/html",
+            DocumentMediaType::CSS => "text/css",
+            DocumentMediaType::MARKDOWN => "text/markdown",
+            DocumentMediaType::CSV => "text/csv",
+            DocumentMediaType::XML => "text/xml",
+            DocumentMediaType::Javascript => "application/x-javascript",
+            DocumentMediaType::Python => "application/x-python",
+        }
+    }
+}
+
+impl AudioMediaType {
+    pub fn from_mime_type(mime_type: &str) -> Option<Self> {
+        match mime_type {
+            "audio/wav" => Some(AudioMediaType::WAV),
+            "audio/mp3" => Some(AudioMediaType::MP3),
+            "audio/aiff" => Some(AudioMediaType::AIFF),
+            "audio/aac" => Some(AudioMediaType::AAC),
+            "audio/ogg" => Some(AudioMediaType::OGG),
+            "audio/flac" => Some(AudioMediaType::FLAC),
+            _ => None,
+        }
+    }
+
+    pub fn to_mime_type(&self) -> &'static str {
+        match self {
+            AudioMediaType::WAV => "audio/wav",
+            AudioMediaType::MP3 => "audio/mp3",
+            AudioMediaType::AIFF => "audio/aiff",
+            AudioMediaType::AAC => "audio/aac",
+            AudioMediaType::OGG => "audio/ogg",
+            AudioMediaType::FLAC => "audio/flac",
+        }
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum ImageDetail {
+    Low,
+    High,
+    Auto,
+}
+
+impl Default for ImageDetail {
+    fn default() -> Self {
+        ImageDetail::Auto
+    }
+}
+
+impl std::str::FromStr for ImageDetail {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "low" => Ok(ImageDetail::Low),
+            "high" => Ok(ImageDetail::High),
+            "auto" => Ok(ImageDetail::Auto),
+            _ => Err(()),
+        }
+    }
+}
+
+impl From<String> for Message {
+    fn from(text: String) -> Self {
+        Message::User {
+            content: OneOrMany::<UserContent>::one(UserContent::Text { text }),
+        }
+    }
+}
+
+impl From<&str> for Message {
+    fn from(text: &str) -> Self {
+        Message::User {
+            content: OneOrMany::<UserContent>::one(UserContent::Text {
+                text: text.to_owned(),
+            }),
+        }
+    }
+}
+
+impl Message {
+    pub fn rag_text(&self) -> Option<String> {
```

Comment:
Make this `pub(crate)`, or use another name that is less use-case specific (maybe `as_str` and return a reference)

Reply:
Also not a fan of this method, but `pub(crate)` with a todo will suffice